driver/src/main/java/org/neo4j/driver/internal/logging/DelegateLogger.java

```diff
@@ -0,0 +1,425 @@
+            break;
+        case INFO:
+            info( msg );
+            break;
+        case WARN:
+            warn( msg );
+            break;
+        case ERROR:
+            error( msg );
+            break;
+        }
+    }
+
+    @Override
+    public void log( InternalLogLevel level, String format, Object arg )
+    {
+        switch ( level )
+        {
+        case TRACE:
+            trace( format, arg );
+            break;
+        case DEBUG:
+            debug( format, arg );
+            break;
+        case INFO:
+            info( format, arg );
+            break;
+        case WARN:
+            warn( format, arg );
+            break;
+        case ERROR:
+            error( format, arg );
+            break;
+        }
+    }
+
+    @Override
+    public void log( InternalLogLevel level, String format, Object argA, Object argB )
+    {
+        switch ( level )
+        {
+        case TRACE:
+            trace( format, argA, argB );
+            break;
+        case DEBUG:
+            debug( format, argA, argB );
+            break;
+        case INFO:
+            info( format, argA, argB );
+            break;
+        case WARN:
+            warn( format, argA, argB );
+            break;
+        case ERROR:
+            error( format, argA, argB );
+            break;
+        }
+    }
+
+    @Override
+    public void log( InternalLogLevel level, String format, Object... arguments )
+    {
+        switch ( level )
+        {
+        case TRACE:
+            trace( format, arguments );
+            break;
+        case DEBUG:
+            debug( format, arguments );
+            break;
+        case INFO:
+            info( format, arguments );
+            break;
+        case WARN:
+            warn( format, arguments );
+            break;
+        case ERROR:
+            error( format, arguments );
+            break;
+        }
+    }
+
+    @Override
+    public void log( InternalLogLevel level, String msg, Throwable t )
+    {
+        switch ( level )
+        {
+        case TRACE:
+            trace( msg, t );
+            break;
+        case DEBUG:
+            debug( msg, t );
+            break;
+        case INFO:
+            info( msg, t );
+            break;
+        case WARN:
+            warn( msg, t );
+            break;
+        case ERROR:
+            error( msg, t );
+            break;
+        }
+    }
+
+    @Override
+    public void log( InternalLogLevel level, Throwable t )
+    {
+        switch ( level )
+        {
+        case TRACE:
+            trace( t );
+            break;
+        case DEBUG:
+            debug( t );
+            break;
+        case INFO:
+            info( t );
+            break;
+        case WARN:
+            warn( t );
+            break;
+        case ERROR:
+            error( t );
+            break;
+        }
+    }
+
+    private String toDriverLoggerFormat( String format )
+    {
+        return format.replaceAll( "\\{\\}", "%s" );
```

Comment:
It might be better for performance to precompile pattern with `Pattern.compile("\\{\\}")` into a `static final` constant and then use it like `PATTERN.matcher(format).replaceAll("%s")`.